src/fides/api/api/v1/endpoints/privacy_experience_endpoints.py

```diff
@@ -259,7 +285,11 @@ async def privacy_experience_list(
 
         results.append(privacy_experience)
 
-    return fastapi_paginate(results, params=params)
+    # This is structured to look like a paginated result to minimize impact from
+    # the caching changes
+    api_result = {"items": results, "total": len(results)}
+    PRIVACY_EXPERIENCE_CACHE[cache_hash] = api_result
```

Comment:
Just want to comment early here.  I see you're caching the output of the entire endpoint: now the performance gains make more sense. This is a neat strategy, I'm just more concerned about accuracy now.  I had been assuming we'd instead cache some time-consuming individual pieces that go into the experience like the output of `get_tcf_contents`. 

For example, `fides_user_device_id` is a unique user identifier.  If present, we supplement the experience with previously-saved preferences for that user.  So if the user saves new preferences, this experience would still return their outdated preferences.

Reply:
That makes sense! Without knowledge of what is actually going on here I can't make those fine-grained caching decisions.

So we can either handle this on the front-end using the header that resets the cache, or we can use a more fine-grained caching method in which I'd need some more guidance on what is safe to cache.

Reply:
real quick, something like the output of `get_tcf_contents` which is the bulk of TCF experiences and PrivacyExperience.get_related_privacy_notices which is for non-TCF experiences but perhaps the latter refactored into a separate method that isn't taking fides_user_device_id in, that is added separately

Reply:
Just a note here, I'm moving a great deal of privacy preferences / privacy experiences endpoints, supporting methods, TCF experiences, etc. to Fidesplus - 
https://ethyca.atlassian.net/browse/PROD-1258

Reply:
thank you for the heads up! In that case I'll move this work over there. I haven't been able to circle back to this yet but Adam already figured it out for some other endpoints we have :)